test/test-framework-spec.md

```diff
@@ -0,0 +1,857 @@
+<ul>
+ 
+<li>A list of function names on which the function depends on. These will be run before the test.
+</li> 
+</ul>
+
+<li>dataProvider : “&lt;function name&gt;” 
+<ul>
+ 
+<li>Specifies the name of the function that will be used to provide the value sets to execute the test against
+</li> 
+</ul>
+
+<li>groups : [“&lt;test group name&gt;”, ...] 
+<ul>
+ 
+<li>List of group names, one or more, that this test belongs to
+</li> 
+</ul>
+</li> 
+</ul>
+   </td>
+  </tr>
+  <tr>
+   <td>@test:AfterSuite {}
+   </td>
+   <td>The function specified in the following annotation will be run once after all the tests in the test suite is run.
+   </td>
+  </tr>
+</table>
+
+### Assertions
+
+The Ballerina test framework has built-in assertions that enable users to verify an actual  output against an expected output. 
+
+The following are the list of available assertions available in the test framework.
+
+<table>
+  <tr>
+   <td>@test:assertEquals
+   </td>
+   <td>Checks if the specified value is equal to the expected value
+   </td>
+  </tr>
+  <tr>
+   <td>@test:assertNotEquals
+   </td>
+   <td>Checks if the specified value is not equal to the expected value
+   </td>
+  </tr>
+  <tr>
+   <td>@test:assertTrue
+   </td>
+   <td>Checks if the specified value is true
+   </td>
+  </tr>
+  <tr>
+   <td>@test:assertFalse
+   </td>
+   <td>Checks if the specified value is false
+   </td>
+  </tr>
+  <tr>
+   <td>@test:assertFail
+   </td>
+   <td>Forces a test case to fail
+   </td>
+  </tr>
+</table>
+
+
+Each assertion allows providing an optional assertion fail message. 
+
+**Example**
+
+
+```ballerina
+@test:Config {}
+function testAssertIntEquals() {
+    int answer = 0;
+    int a = 5;
+    int b = 3;
+    answer = intAdd(a, b);
+    test:assertEquals(answer, 8, msg = "int values not equal");
+}
+```
+
+
+### Grouping Tests
+
+Organizing tests into groups allows the user to run a specific group of tests using the `--groups` option. 
+
+**Example**
+
+```ballerina
+@test:Config { groups: ['group1'] }
+function test1() {
+    // Test code ...
+}
+```
+
+### Enabling / Disabling Tests
+
+Individual test cases can be disabled by adding the `disabled` option in the test config annotation. This makes sure that the particular test case is skipped when running the tests. 
+
+```ballerina
+@test:Config { enable : false }
+function test1() {
+    // Test code ...
+}
+```
+
+When a test case is disabled, before and after functions specified in the test configurations and all the dependent
+ test cases will be skipped. 
+
+
+### Order of Execution
+
+The depends on attribute can be used to define a list of function names that the test function depends on. These functions will be executed before the test execution. The order in which the comma-separated list appears has no prominence and thus will be executed in an arbitrary manner. This attribute can be used to ensure that the tests are being executed in the expected order.
```

Comment:
Shall we change the order of last 2 sentences? Instead of "the comma separated list", shall we use "comma separated list of functions" ?